pub/lib/dependabot/pub/update_checker.rb

```diff
@@ -42,26 +42,31 @@ def lowest_resolvable_security_fix_version
       end
 
       def lowest_security_fix_version
-        # TODO: Pub lacks a lowest-non-vulnerable version strategy, for now we simply bump to latest resolvable:
-        # https://github.com/dependabot/dependabot-core/issues/5391
-        relevant_version = latest_resolvable_version
-        return unless relevant_version
-
-        # NOTE: in other ecosystems, the native helpers return a list of possible versions, to which we apply
-        # post-filtering. Ideally we move toward a world where we hand the native helper a list of ignored versions
-        # and possibly a flag indicating "use min version rather than max". The pub team is interested in supporting
-        # that. But in the meantime for internal consistency with other dependabot ecosystem implementations I kept
-        # `relevant_versions` as an array.
-        relevant_versions = [relevant_version]
-        relevant_versions = Dependabot::UpdateCheckers::VersionFilters.filter_vulnerable_versions(relevant_versions,
-                                                                                                  security_advisories)
-        relevant_versions.min
+        # Don't attempt to do security updates for git dependencies.
+        return nil if git_revision? dependency.version
+        # If the current version is not vulnerable, we stay on it.
+        return version_unless_ignored dependency.version unless vulnerable?
+
+        e = dependency_services_smallest_update
+        return nil if e.nil?
+
+        upgrade = e.find { |u| u["name"] == dependency.name }
+
+        version = upgrade["version"]
+        version_unless_ignored(version)
       end
 
       def updated_requirements
         # Requirements that need to be changed, if obtain:
-        # latest_resolvable_version
-        entry = current_report["singleBreaking"].find { |d| d["name"] == dependency.name }
+        # latest_resolvable_version or lowest_security_fix_version
+        entry = if vulnerable?
+                  updates = dependency_services_smallest_update
+                  raise "Cannot upgrade from vulnerability without unlock." if update.size > 1
```

Comment:
Tried this against a repo and run into an error due to a typo here:

```suggestion
                  raise "Cannot upgrade from vulnerability without unlock." if updates.size > 1
```

I guess we should get this covered, maybe through a smoke test at https://github.com/dependabot/smoke-tests.

Reply:
Thanks for catching this!

Will you be able to set up a smoke test?

Reply:
Not right now, but I'll try set that up!